subsys/bluetooth/mesh/friend.c

```diff
@@ -1402,7 +1402,7 @@ static void friend_lpn_enqueue_rx(struct bt_mesh_friend *frnd,
 	 * this rx function. These packets have already been added to the
 	 * queue, and should be ignored.
 	 */
-	if (bt_mesh_elem_find(rx->ctx.addr)) {
+	if (bt_mesh_has_addr(rx->ctx.addr, false)) {
```

Comment:
Since we're checking the source address here, the group address check is never actually relevant. With this in mind, the only need we have for the `trigger_msg_cb` flag is for the loopback. If we forward the loopback packets to the callback as well, we can remove this flag.

Reply:
Yes, in that case we wouldn't need this flag anymore. Since I am indeed going to forward loopback packets to the callback I can remove this flag.